rest/server_context.go

```diff
@@ -1117,27 +1132,25 @@ func initClusterAgent(clusterAddress, clusterUser, clusterPass, certPath, keyPat
 	return agent, nil
 }
 
-func (sc *ServerContext) ObtainManagementEndpointsAndHTTPClient() ([]string, *http.Client, error) {
+// Obtains a gocb agent from the current server connection. Requires the agent to be closed after use
+func (sc *ServerContext) initializeGoCBHttpClient() (*gocbcore.Agent, error) {
 	agent, err := initClusterAgent(
 		sc.config.Bootstrap.Server, sc.config.Bootstrap.Username, sc.config.Bootstrap.Password,
 		sc.config.Bootstrap.X509CertPath, sc.config.Bootstrap.X509KeyPath, sc.config.Bootstrap.CACertPath, sc.config.Bootstrap.ServerTLSSkipVerify,
 		sc.config.API.ServerReadTimeout.Value())
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	managementEndpoints := agent.MgmtEps()
-
-	httpClient := &http.Client{
-		Transport: agent.HTTPClient().Transport,
-	}
+	return agent, nil
+}
 
-	err = agent.Close()
-	if err != nil {
-		return nil, nil, err
+func (sc *ServerContext) ObtainManagementEndpointsAndHTTPClient() ([]string, *http.Client, error) {
+	if sc.GoCBAgent == nil {
+		return nil, nil, fmt.Errorf("unable to obtain agent. Running with Walrus? ")
```

Comment:
I don't think the 'Running with Walrus?' error message is great, if this can make it back to the end user, since presumably we should know if they are running with Walrus.